x-pack/plugins/maps/public/actions/data_request_actions.ts

```diff
@@ -365,6 +366,9 @@ export function fitToDataBounds() {
     }
 
     if (!corners.length) {
+      if (onNoBounds) {
```

Comment:
Added this callback to fix error with autofit to bounds where when there are no matching documents, sync all layers is not triggered.

Reply:
The problem is that the method is a redux thunk. I do not believe thunks can return values. All state gets updated via dispatch so returns can just be wither a function (another thunk) or pure redux action (like `{ type: SET_MAP_SETTINGS, settings }`) .